src/System.Management.Automation/engine/runtime/Binding/Binders.cs

```diff
@@ -7199,17 +7203,26 @@ internal static Expression InvokeMethod(MethodBase mi, DynamicMetaObject target,
                         var psRefValue = Expression.Property(args[i].Expression.Cast(typeof(PSReference)), CachedReflectionInfo.PSReference_Value);
                         initTemps.Add(Expression.Assign(temp, psRefValue.Convert(temp.Type)));
                         copyOutTemps.Add(Expression.Assign(psRefValue, temp.Cast(typeof(object))));
+
                         argExprs[i] = temp;
+                        // Log the PSReference type, or its actual value in this case?
+                        argsToLog.Add(args[i].Expression);
```

Comment:
@TravisEz13 @SeeminglyScience For by-ref parameters (`out` or `ref` parameters) that we pass in a `PSReference` instance, shall we simply log the `PSReference` type or the value the instance holds? For methods with `out` parameters, the value is usually `null`.

Reply:
> @TravisEz13 @SeeminglyScience For by-ref parameters (`out` or `ref` parameters) that we pass in a `PSReference` instance, shall we simply log the `PSReference` type or the value the instance holds? For methods with `out` parameters, the value is usually `null`.

So I'm thinking we want to log the value. I don't think we're really worried about logging `out` parameters, that's like saying logging "we are going to get a value".

For `ref` and `in` parameters, the value is what is meaningful.

Reply:
@TravisEz13 Your thoughts?

Reply:
Updated to log the value of passed-in `PSReference` object.